### PR TITLE
Feature/hybrid switch players

### PR DESCRIPTION
**Pull request by DJSteins**
Implements hybrid switch players.
- 0 - 1 Players: do nothing
- 2 Players: Switch the two players
- 3 Players: Switch the two players who did not use the power up
- 4 Players: Fischer-Yates shuffle to randomly shuffle the 3 players who did not use the power up

### CODE REVIEW

**Comment by mikeslgoh** [comment]
[Implement hybrid switch player powerup](https://trello.com/c/ouaiodOj/493-implement-hybrid-switch-player-powerup)

File: Assets/_PartyParkade/Script/Game/DebugInventory.cs [hunk 1]
```diff
@@ -36,9 +36,7 @@ private void BindPowerUps()
                 {KeyCode.Alpha2, POWERUPS.GravityWell},
                 {KeyCode.Alpha3, POWERUPS.OilSpill},
                 {KeyCode.Alpha4, POWERUPS.Boost},
-                {KeyCode.Alpha5, POWERUPS.SwitchPlayers},
-                {KeyCode.Alpha6, POWERUPS.SwitchAllPlayers},
-                {KeyCode.Alpha7, POWERUPS.SwitchTwoPlayersRandomly}
+                {KeyCode.Alpha6, POWERUPS.SwitchAllPlayers}
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/DebugInventory.cs:39]: do we still need this class? since we have god mode?

Comment (mikeslgoh) [reply Assets/_PartyParkade/Script/Game/DebugInventory.cs:39]: I think we can delete it and remove it from the player prefab

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Game/DebugInventory.cs:39]: Okay can you do that?

File: Assets/_PartyParkade/Script/PowerUp/SwitchAllPlayers.cs [hunk 2]
```diff
@@ -4,29 +4,59 @@
...
-            PlayerManager.Players[k].Car.SetCarPosition(PlayerManager.Players[n].Car.GetCarPosition());  
-            // set the position of the car we are on
-            PlayerManager.Players[n].Car.SetCarPosition(position);
-        }  
+        if (n == 1)
+        {
+            // switch the two players in the game
+            ShuffleTwo(playerController, controllers[0]);
+        }
+        else if (n == 2)
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/PowerUp/SwitchAllPlayers.cs:40]: Should this be converted to switch cases?

File: Assets/_PartyParkade/Script/PowerUp/SwitchAllPlayers.cs [hunk 3]
```diff
@@ -4,29 +4,59 @@
...
+            // switch the two players that are not the player who used the power up
+            ShuffleTwo(controllers[0], controllers[1]);
+        }
+        else
+        { 
+            // randomly switch all players who are not the player who used the power up
+            // this may or may not switch all other players - it's random
+            while (n > 1)
+            {
+                n--;
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/PowerUp/SwitchAllPlayers.cs:51]: maybe put this in separate method

File: Assets/_PartyParkade/Script/Game/DebugInventory.cs (hunk 1 above)
Comment (mikeslgoh) [reply Assets/_PartyParkade/Script/Game/DebugInventory.cs:39]: I'll wait for Daniel to make the changes and remove this before merging it

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Game/DebugInventory.cs:39]: @mikeslgoh oh. idk why i thought u were daniel replying. 

i meant Daniel should make the changes

File: Assets/_PartyParkade/Script/PowerUp/SwitchAllPlayers.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/_PartyParkade/Script/PowerUp/SwitchAllPlayers.cs:40]: I agree

File: Assets/_PartyParkade/Script/Game/DebugInventory.cs (hunk 1 above)
Comment (mikeslgoh) [reply Assets/_PartyParkade/Script/Game/DebugInventory.cs:39]: haha ok that's what I thought too

**Pull request merged**